scripts/09-history-migration-to-mongo.py

```diff
@@ -0,0 +1,455 @@
+        group_history._dirty = True
+        return group_history
+
+    def dump(self):
+        return {
+            'group_id': self.group_id,
+            'couples': [cr.dump() for cr in self.couples],
+            'nodes': [nbrs.dump() for nbrs in self.nodes],
+        }
+
+    @staticmethod
+    def list(collection, **kwargs):
+        """
+        TODO: This should be moved to some kind of Mongo Session object
+        """
+        params = {}
+
+        for k, v in kwargs.iteritems():
+            if v is None:
+                continue
+            params.update(condition(k, v))
+
+        return collection.find(params)
+
+
+def condition(field_name, field_val):
+    if isinstance(field_val, (list, tuple)):
+        return {field_name: {'$in': list(field_val)}}
+    else:
+        return {field_name: field_val}
+
+
+class GroupNodeBackendsSet(list):
+    def __init__(self, *args, **kwargs):
+        super(GroupNodeBackendsSet, self).__init__(*args, **kwargs)
+        if len(self) != len(set(*args, **kwargs)):
+            raise ValueError('Node backends set should contain only unique elements')
+
+    def append(self, item):
+        if item in self:
+            raise ValueError('Node backend {} is already in node backends set')
+        super(GroupNodeBackendsSet, self).append(item)
+
+    def extend(self, items):
+        for item in items:
+            self.append(item)
+
+    def insert(self, idx, item):
+        if item in self:
+            raise ValueError('Node backend {} is already in node backends set')
+        super(GroupNodeBackendsSet, self).insert(idx, item)
+
+    def __setitem__(self, idx, item):
+        raise NotImplementedError('GroupNodeBackends set does not support __setitem__ method')
+
+    def __setslice__(self, i, j, items):
+        raise NotImplementedError('GroupNodeBackends set does not support __setslice__ method')
+
+    def __getslice__(self, i, j):
+        return GroupNodeBackendsSet(super(GroupNodeBackendsSet, self).__getslice__(i, j))
+
+    def __add__(self, other):
+        obj = GroupNodeBackendsSet()
+        for nb in self:
+            obj.append(nb)
+        for nb in other:
+            obj.append(nb)
+        return obj
+
+
+class GroupNodeBackendsSetRecord(object):
+    # TODO: Move constats to history module
+    def __init__(self, set, timestamp=None, type='automatic'):
+        self.set = GroupNodeBackendsSet(
+            nbr if isinstance(nbr, GroupNodeBackendRecord) else GroupNodeBackendRecord(**nbr)
+            for nbr in set
+        )
+        self.timestamp = timestamp
+        self.type = type
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __eq__(self, other):
+        def make_cast(s):
+            cast = {}
+            for nbr in s.set:
+                cast.setdefault(nbr, 0)
+                cast[nbr] += 1
+            return cast
+        c1 = make_cast(self)
+        c2 = make_cast(other)
+        return c1 == c2
```

Comment:
Looks like you forgot to update this code.

Why did you even include this script in the pull request?